### PR TITLE
AlignmentMatrixControl: Promote to stable

### PR DESCRIPTION
**Pull request by fullofcaffeine**
Part of #59418

## What?

Promote `AlignmentMatrixControl` to a non-experimental component.

## Why?

The strategy of prefixing exports with `__experimental` has become deprecated after the introduction of private APIs. 

## Checklist

- [x] Export it from components without the `__experimental` prefix.
- [x] Switch subcomponents to dot notation if applicable.
- [x] Keep the old `__experimental` export for backwards compatibility.
- [x] Remove experimental callout from readme.
- [x] Change all imports of the old `__experimental` in GB and components to the one without the prefix (including in Storybook stories).
- [x] Update the docs (readme, JSDocs, Storybook) to refer to the new unprefixed component.
- [x] Add the component storybook `id` (get it from the storybook URL) to the `PREVIOUSLY_EXPERIMENTAL_COMPONENTS` const array in `manager-head.html` so that old experimental story paths are redirected to the new one.
- [x] Add a changelog for both the promotion (enhancement) and the deprecation.

## ✍️  Dev note

The `__experimentalAlignmentMatrixControl` component can now be imported as `AlignmentMatrixControl`.

The legacy `__experimentalAlignmentMatrixControl` is marked as deprecated.

### CODE REVIEW

File: storybook/manager-head.html [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <script>
 	( function redirectIfStoryMoved() {
 		const PREVIOUSLY_EXPERIMENTAL_COMPONENTS = [
-			'navigation',
+			'alignmentmatrixcontrol',
 			'customselectcontrol-v2',
-			'theme',
+			'navigation',
 			'progressbar',
+			'theme',
```
Comment (mirka) [inline storybook/manager-head.html:8]: Sorted alphabetically

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -357,6 +358,7 @@
 ### Deprecations
 
 -   `Navigation`: Soft deprecate component ([#59182](https://github.com/WordPress/gutenberg/pull/59182)).
+-   `AlignmentMatrixControl`: Remove "experimental" designation ([#60913](https://github.com/WordPress/gutenberg/pull/60913)).
```
Comment (tyxla) [inline packages/components/CHANGELOG.md]: Seems like we're adding this to the wrong release. Should this go up in its own new **Deprecations** section?

File: packages/components/src/alignment-matrix-control/index.tsx [hunk 3]
```diff
@@ -100,6 +80,41 @@ export function AlignmentMatrixControl( {
...
+		 *
+		 * ```jsx
+		 * import { AlignmentMatrixControl } from '@wordpress/components';
+		 *
+		 * <Icon icon={<AlignmentMatrixControl.Icon value="top left" />} />
+		 * ```
+		 */
+		Icon: Object.assign( AlignmentMatrixControlIcon, {
+			displayName: 'AlignmentMatrixControl.Icon',
+		} ),
```
Comment (tyxla) [inline packages/components/src/alignment-matrix-control/index.tsx:116]: 👍

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Thanks, @fullofcaffeine, for the initial work and for getting it to the finish line @mirka! 🚀

File: packages/components/src/index.ts [hunk 4]
```diff
@@ -12,7 +12,11 @@ export {
 } from '@wordpress/primitives';
 
 // Components.
-export { default as __experimentalAlignmentMatrixControl } from './alignment-matrix-control';
+export {
+	/** @deprecated Import `AlignmentMatrixControl` instead. */
```
Comment (ciampo) [inline packages/components/src/index.ts:16]: Do we plan on adding a deprecation warning in a follow-up once all instances are migrated, or are we keeping it "soft deprecated" for now?

Comment (mirka) [reply packages/components/src/index.ts:16]: I'm thinking we can decide on a per-component basis, using reasoning from #61099. Maybe we'll want to do a more aggressive cleanup in the future, but I don't think it's a pressing need for the moment. I'm anticipating most of these to be passive deprecations.

File: packages/components/CHANGELOG.md (hunk 2 above)
Comment (mirka) [reply packages/components/CHANGELOG.md]: Fixed 🙈

**Comment by tyxla** [comment]
Seems like this got conflicted and will need a rebase.

**Pull request merged**